Tests/Realm.Tests/Sync/StaticQueriesTests.cs

```diff
@@ -0,0 +1,1257 @@
+
+                    var fullyRetrieved = await WaitForConditionAsync(() => collection.FindOneAsync(new { _id = original.Id }), item => Task.FromResult(item != null));
+
+                    Assert.That(fullyRetrieved.Id, Is.EqualTo(original.Id));
+                    Assert.That(fullyRetrieved.Value, Is.EqualTo(original.Value));
+                }
+            }, timeout: 120000);
+        }
+
+        public static readonly object[] RealmValueLinkTestCases = new[]
+        {
+            new object[]
+            {
+                CreateTestCase("Single link", new RealmValueObject
+                {
+                    RealmValueProperty = new IntPropertyObject { Int = 2 },
+                }),
+            },
+            new object[]
+            {
+                CreateTestCase("List", new RealmValueObject
+                {
+                    RealmValueList =
+                    {
+                        new IntPropertyObject { Int = 100 },
+                        new IntPropertyObject { Int = 200 },
+                    },
+                }),
+            },
+            new object[]
+            {
+                CreateTestCase("Dictionary", new RealmValueObject
+                {
+                    RealmValueDictionary =
+                    {
+                        ["key_1"] = new IntPropertyObject { Int = 100 },
+                        ["key_null"] = RealmValue.Null,
+                        ["key_2"] = new IntPropertyObject { Int = 200 },
+                    },
+                })
+            },
+            new object[]
+            {
+                CreateTestCase("Set", new RealmValueObject
+                {
+                    RealmValueSet =
+                    {
+                        new IntPropertyObject { Int = 100 },
+                        new IntPropertyObject { Int = 200 },
+                    },
+                }),
+            },
+            new object[]
+            {
+                CreateTestCase("All types", new RealmValueObject
+                {
+                    RealmValueProperty = new IntPropertyObject { Int = 2 },
+                    RealmValueList =
+                    {
+                        new IntPropertyObject { Int = 3 },
+                        new IntPropertyObject { Int = 4 },
+                    },
+                    RealmValueSet =
+                    {
+                        new IntPropertyObject { Int = 5 },
+                        new IntPropertyObject { Int = 6 },
+                    },
+                    RealmValueDictionary =
+                    {
+                        ["dict_1"] = new IntPropertyObject { Int = 7 },
+                        ["dict_2"] = new IntPropertyObject { Int = 8 },
+                        ["dict_null"] = RealmValue.Null
+                    }
+                }),
+            }
+        };
+
+        // TODO This is going to be fixed with https://jira.mongodb.org/browse/BAAS-27410
```

Comment:
That ticket seems to be resolved, but perhaps it's not deployed to QA yet?

Reply:
Exactly.